src/components/Form/AllowedValueComboBox.vue

```diff
@@ -58,33 +83,78 @@ const props = defineProps({
     type: String,
     required: true,
   },
-  isFetched: Boolean,
+  isFetching: Boolean,
 });
 
-const emit = defineEmits(["update:modelValue"]);
+const sku = ref("");
+const isSkuLocked = ref(true);
+
+const onValueChange = (val) => {
+  const found = props.items.find((item) => item.value === String(val));
+
+  if (found) {
+    sku.value = found?.sku;
+    isSkuLocked.value = true;
+  } else {
+    sku.value = "";
+    isSkuLocked.value = false;
+  }
+
+  updateModelValue(val);
+  updateAllowedValueDetails(val);
+};
+
+const onSkuChange = (val) => {
+  sku.value = val;
+  updateAllowedValueDetails(props.modelValue);
+};
+
+const updateModelValue = (val) => {
+  emit("update:modelValue", val);
+};
+
+const updateAllowedValueDetails = (val) => {
+  emit("update:allowedValueDetails", {
+    value: val,
+    sku: sku.value,
+  });
+};
+
+const emit = defineEmits(["update:modelValue", "update:allowedValueDetails"]);
+
+// Reinitialize allowed value state whenever modelValue or isFetching change.
+// This handles cases like form resets or loading existing data (edit mode),
+// ensuring modelValue and related SKU fields stay in sync.
+watch(
+  [() => props.modelValue, () => props.isFetching],
+  ([newModelValue]) => {
+    if (newModelValue) {
+      onValueChange(newModelValue);
+    } else sku.value = "";
+  },
+  { immediate: true }
+);
+
+const applySkuRules = () => {
+  return props.modelValue ? props.rules : [];
+};
 
 const store = useStore();
 
 const deleteDialog = ref(false);
 const valueToDelete = ref("");
 
 function confirmDelete(value) {
-  const stringValue =
-    typeof value === "object" && value !== null
-      ? value.value || value.title || value
-      : value;
-
-  valueToDelete.value = stringValue;
+  valueToDelete.value = props.items.find((v) => v.value === value.value);
```

Comment:
v.value === value.value is obscure.  
props.items could become props.storedAllowedValues
confirmDelete(value) -> confirmDelete(selected)
```suggestion
  valueToDelete.value = props.storedAllowedValues.find((stored) => stored.value === selected.value);
```
Other suggestions are also welcome